Ricochet/Assets/_Scripts/PowerUp.cs

```diff
@@ -31,16 +44,33 @@ private void OnTriggerEnter2D(Collider2D collider)
                 gameManagerInstance.PlayerPowerUpCollision(collider.gameObject, this);
                 gameManagerInstance.RespawnPowerUp(gameObject);
                 gameObject.SetActive(false);
-                
             }
         }
     }
+
+    void OnEnable()
+    {
+        if (powerUpType == EPowerUp.Random)
+        {
+            instanceType = powerups[rng.Next(powerups.Count)];
+            Debug.Log("POWER UP SPAWNED: " + instanceType);
+            UpdateSprite();
+        }
+    }
     #endregion
 
     #region External Functions
     public EPowerUp GetPowerUpType()
     {
-        return powerUpType;
+        return instanceType;
+    }
+
+    private void UpdateSprite()
```

Comment:
This should be moved into a Private Functions region since it is not externally accessible.